packages/ra-core/src/createAdminStore.ts

```diff
@@ -65,15 +65,21 @@ export default ({
     const sagaMiddleware = createSagaMiddleware();
     const typedWindow = window as Window;
 
+    const composeEnhancers =
+        (process.env.NODE_ENV === 'development' &&
+            typeof typedWindow !== 'undefined' &&
+            typedWindow.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
+            typedWindow.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
+                trace: true,
+                traceLimit: 25,
+            })) ||
+        compose;
```

Comment:
I don't understand this code. It looks like you're changing the `compose` method but you're not, and using `compose` isn't even necessary since there is only one middleware.

Reply:
Is a suggestion from [this post: [When browser doesn't have the redux devtools extension the app throws an undefined error](https://github.com/zalmoxisus/redux-devtools-extension/issues/320)

Reply:
ah, sorry, I misread the change:  `typedWindow.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()` actually returns a `compose` function. Now it makes sense.

Reply:
I've updated "CustomApp.md" file.